src/components/common/DropDown.tsx

```diff
@@ -0,0 +1,106 @@
+import { FC, useRef, useState, useEffect } from 'react';
+
+import useOuterClick from '@/hooks/useOuterClick';
+
+export interface IDropdownOption {
+  value: string;
+  label: string;
+}
+
+interface IDropdownProps {
+  options: IDropdownOption[];
+  value?: string;
+  onChange?: (value: string) => void;
+  bgColor?: string;
+  textColor?: string;
+  iconColor?: string;
+  placeholder?: string;
+}
+
+/*아래는 미지정 시 기본값이며 이 컴포넌트 사용하실 때마다 디자인 맞춰서 바꿔주시면 됩니다!*/
+const Dropdown: FC<IDropdownProps> = ({
```

Comment:
P3)

나중에 enum 같은 거 넣을 때 좀더 type safe 하게 동작할 수 있게 하려면 이게 좀더 낫습니다

FC 쓰면 이런 식으로 타입 지정을 못 한다는 단점이 있어서, 위에서도 언급했듯 FC는 되도록 안 쓰는 게 좋습니다

```suggestion
export interface IDropdownOption<T extends string> {
  value: T;
  label: string;
}

interface IDropdownProps<T extends string> {
  options: IDropdownOption<T>[];
  value?: T;
  onChange?: (value: T) => void;
  bgColor?: string;
  textColor?: string;
  iconColor?: string;
  placeholder?: string;
}

/*아래는 미지정 시 기본값이며 이 컴포넌트 사용하실 때마다 디자인 맞춰서 바꿔주시면 됩니다!*/
const Dropdown = <T extends string>({ 프롭들 }: DropdownOption<T>) => 
```

Reply:
@woohm402 현민님 드롭다운 사용중 질문이 생겼는데 value 로 number 를 쓰고 싶은데 T 를 extends string 으로 해주신 이유 있을까요??

Reply:
@khinwaiyan 아 string 으로 써주셨길래 string 으로 했어요! `T extends number` 도 되고 `T extends string | number` 로 해도 상관없습니다 👍